### PR TITLE
Refactor url and add medium to the attribute call

### PR DESCRIPTION
**Pull request by gtupak**
The `url` is used internally to track page changes, but I don't se a reason to store it on the window object, instead moving it to the class.

Also, the `attribute()` call should also support a `medium`, as it is commonly used among utm tags.

### CODE REVIEW

**Review by kQuintal — APPROVED** [review]
You simply added a top-level variable to the `ARCxAnalyticsSDK` class.

LGTM

**Comment by gtupak** [comment]
Made some modifications to the original PR:
- Instead of storing the `currentURL` in `window` or in the sdk's instance, it is now stored in the `sessionStorage`, to avoid situations where bad implementations of the SDK triggers repetitive `REFERRER` calls
- Improved tests by using `jsdom` and checking the `window`'s  `referrer`, `location`, etc

**Comment by xitronix** [comment]
Test notifications

File: test/ArcxAnalyticsSdk.test.ts [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-import { CONNECT_EVENT, PAGE_EVENT, TRANSACTION_EVENT, DEFAULT_SDK_CONFIG } from '../src/constants'
+import globalJsdom from 'global-jsdom'
+import {
+  CONNECT_EVENT,
+  PAGE_EVENT,
+  TRANSACTION_EVENT,
+  DEFAULT_SDK_CONFIG,
+  CURRENT_URL_KEY,
+  REFERRER_EVENT,
+  // REFERRER_EVENT,
```
Comment (kQuintal) [inline test/ArcxAnalyticsSdk.test.ts:13]: 🎯

**Review by kQuintal — APPROVED** [review]
Lgtm

File: src/ArcxAnalyticsSdk.ts [hunk 2]
```diff
@@ -28,14 +29,33 @@ export class ArcxAnalyticsSdk {
...
     document.body.addEventListener(
       'click',
       () => {
         requestAnimationFrame(() => {
-          if (window.url !== location.href) {
-            window.url = location.href
-            this.page({ url: window.url })
+          if (this.shouldFirePageEvent()) {
+            this.page({ url: location.href })
           }
```
[the remark this answers is not in the document]
Comment (gtupak) [reply src/ArcxAnalyticsSdk.ts]: I like this, updated

**Pull request merged**